src/main/java/com/iota/iri/cache/Cache.java

```diff
@@ -0,0 +1,117 @@
+package com.iota.iri.cache;
+
+import com.iota.iri.cache.util.ValueWrapper;
+import com.iota.iri.storage.PersistenceProvider;
+
+import java.util.Collection;
+import java.util.Map;
+
+public interface Cache {
+
+    /**
+     * Get the name of this cache.
+     * 
+     * @return The name of the cache
+     */
+    String getName();
+
+    /**
+     * Get the value mapped to the specified key. Returns {@code null} if the mapping is not found.
+     * 
+     * @param key The keys whos value is to be returned
+     * @return The mapped value of the specified key
+     */
+    ValueWrapper get(Object key);
+
+    /**
+     * Get the value mapped to the specified key cast to the requested type. This does not differentiate between a
+     * cached {@code null} and no cached value at all. If the value is {@code null}, the return type is therefore
+     * irrelevant. If the value does not match the requested type, an {@code IllegalStateException} is thrown.
+     * 
+     * @param key  The key whos value is to be returned
+     * @param type The type of the value to match
+     * @param <T>  The class type
+     * @return The mapped value of the specified key.
+     */
+    <T> T get(Object key, Class<T> type) throws IllegalStateException;
+
+    /**
+     * Get the value mapped to the specified key obtaining the value from the persistence provider if not found in
+     * cache. If cache return; Otherwise db read, cache and return.
+     * 
+     * @param key                 The key wohs value is to be returned
+     * @param persistenceProvider
+     * @param <T>                 The class type of the returned value
+     * @return The mapped value of the specified key
+     */
+    <T> T get(Object key, PersistenceProvider persistenceProvider);
+
+    /**
+     * Get all elements from the cache for the keys specified.
+     * 
+     * @param keys The keys to get values of
+     * @return A map of key and value associations.
+     */
+    Map<Object, ValueWrapper> getAll(Collection<Object> keys);
```

Comment:
Why don't you start using generics that extend `Indexable` and `Persistable`?
I think you ditch the `ValueWarpper`. 

It is just unnecessary overhead...
We have too many `Transactions Objects` as is